packages/flutter/lib/src/material/page.dart

```diff
@@ -85,59 +96,37 @@ class MaterialPageRoute<T> extends PageRoute<T> {
 
   @override
   bool canTransitionFrom(TransitionRoute<dynamic> nextRoute) {
-    return nextRoute is MaterialPageRoute<dynamic>;
+    return nextRoute is MaterialPageRoute<dynamic> || nextRoute is CupertinoPageRoute<dynamic>;
```

Comment:
Really hoped too that since the analyzer can figure out automatic casting for `a is A && a.somethingADoes`, that it would also do `(a is A || a is B) && a.somethingAAndB'sSuperclassDoes`